clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkClient.java

```diff
@@ -369,24 +393,40 @@ private void maybeTriggerWakeup() {
     }
 
     public void disableWakeups() {
-        wakeupDisabledCount++;
+        lock.lock();
```

Comment:
One general comment: could we wrap this pattern as sth. like we did with Scala's `inlock` to avoid code duplication?